File: Core/Entities/Payable.cs
namespace Core.Entities
{
    public class Payable : BaseEntity
    {
        public decimal AmountDue { get; set; }
        public decimal AmountPaid { get; set; }
        public DateTime DatePaid { get; set; }
        public string Method { get; set; }
        public string Notes { get; set; }
        public int EntityId { get; set; }
        public Entity Entity { get; set; }
        public int GigId { get; set; }
        public Gig Gig { get; set; }
    }
}

File: client/src/app/shared/models/payable.ts
import { IGig } from "./gig";

export interface IPayable {
    id: number;
    amountDue: number;
    amountPaid: number;
    datePaid: Date;
    method: string;
    notes: string;
    entityId: number;
    entity: string;
    gig: IGig;
}